### DISCUSSION TITLE
Stellantis Peugeot e-208 / Opel Corsa

### DISCUSSION THREAD

**Discussion by deadrabbit87** [meatpiHQ/wican-fw#278] [Q&A]
Hi, 

i wan't to get rid of the original Stellantis / My-Peugeot App and use WiCan instead of that. 

As i see, theres no template for this vehicle. What i found is [this](https://www.goingelectric.de/forum/viewtopic.php?p=1514148&sid=038f7b3d9279c69e714d3d0d47a95c73#p1514148) thread in which some IDs are mentioned: 
```
#######
####### Setup commands
#######
ATWS ---> Warm reset set back to defaults
ATI ---> Show interface ID
ATE0 ---> Turn off terminal echo
AT@1 --->
ATSP6 ---> Set to protocol 6: ISO 15765-4 CAN (11 bit ID 500 kbaud)
ATAT1 --->
ATL0 ---> Disable line feed
ATS0 ---> Disable sending spaces between hex characters
ATH1 ---> Turn on odb2 header
ATCAF1 ---> Turn on CAN auto formatting
ATSH79B ---> Set request message header 79B
ATFCSH79B ---> Set flow control header (CAN ID)
ATFCSD300000 ---> Set flow control data
ATFCSM1 ---> Set flow control mode; 1 = User defined CAN ID and data

#######
####### Data commands
#######
ATSH6A2 ---> Set the header of transmitted OBD messages to header 6A2
ATFCSH6A2 ---> Set flow control header (CAN ID)
ATCRA682 ---> Set the CAN hardware filter to 682
22D4341 ---> Ambient temperature
22D8EF1 ---> Battery temperature
22D4021 ---> Speed
ATSH6B4 ---> Set the header of transmitted OBD messages to header 6B4
ATFCSH6B4 ---> Set flow control header (CAN ID)
ATCRA694 ---> Set the CAN hardware filter to 694
22D8161 ---> Battery current
22D8151 ---> Battery voltage
22D86F1 ---> Battery minimum voltage
22D8701 ---> Battery maximum voltage
22D4101 ---> SOC calibrated
22D8651 ---> kWh available
22D8601 ---> SOH
22D8101 ---> SOC
```

Is there a way to create a template for this vehicle with this data?

**Comment by th84sl** [comment]
Hello, I did not create a template, but I have a running WICAN configuration. Please see attached PDF-file.
[get SOC with WICAN from Peugeot 208e v2.pdf](https://github.com/user-attachments/files/18177325/get.SOC.with.WICAN.from.Peugeot.208e.v2.pdf)

**Comment by meatpiHQ** [comment]
@deadrabbit87 @th84sl 

I've created a profile for the e-208, please try it and let me know if it works.

**Reply by th84sl** [reply]
Yes it works, thank you

**Reply by deadrabbit87** [reply]
Sorry for the late reply. I haven't had time to test the profile. 

Today I tested the template while the car was charging. I could not get any valid values. 

CAN is online and responding, but I only get timeouts for the relevant PIDs. 

@th84sl Have you tried this with the ignition on or off?

**Reply by slash909uk** [reply]
Hi all - wanted to follow up on this thread with more confirmation of a working profile. I have just connected a WiCAN-OBD-3 to my Vauxhall Corsa-e and can confirm the e-208 profile works just fine with this model too (expected since they are same car platform.) I can also confirm no adapter cable is required as I have updated my car firmware which enables the standard CAN bus pins (6,14) on the connector.

Now experimenting with exactly what state the car needs to be in to generate battery data - I can get a response for a short time just by pressing the lock button on the remote key fob even if the car is already locked - maybe only odometer data however? I get all data by unlocking the door and of course pressing 'start'. I begin to wonder if there is some CAN bus command to wake up the required ECU module that we can send from the meatPi... clearly the key fob receiver can do exactly this. I will update you on what I find.

**Comment by Dity1** [comment]
These values ​​also apply to the Fiat 600 e and it works.

**Reply by deadrabbit87** [reply]
Also if the ignition is switched off?

**Reply by Dity1** [reply]
Ignition must be on and sometimes there are breaks.
With v400 I couldn't connect to the network. v348 worked again

**Reply by deadrabbit87** [reply]
Hm, with Igntion on it's useless for me. I will use the data from the car while it's charing. 

There is traffic on the bus even when the ignition off.

Does anyone knows if it is possible, to get some data when the ignition is off?

**Reply by meatpiHQ** [reply]
> Ignition must be on and sometimes there are breaks. With v400 I couldn't connect to the network. v348 worked again

@Dity1 is your WiFi network still on WPA2? If so you can use v4.00 and set the security to WPA2.

**Reply by Dity1** [reply]
Okay thanks, I'll test it

**Reply by moritzklein** [reply]
> Hm, with Igntion on it's useless for me. I will use the data from the car while it's charing.
> 
> There is traffic on the bus even when the ignition off.
> 
> Does anyone knows if it is possible, to get some data when the ignition is off?

I have a opel mokka, uses the same EV system as the corsa.
Regarding ignition - that's not a problem for me, because I use evcc to manage charging.
Once I arrive at home, the wican-obd connects to my wifi and send the soc via mqtt to evcc.
Then I park the car and plug it into the EV station. evcc will save the soc and interpolate the soc during charging until it gets an updated soc from the car via mqtt.
Yes the interpolation isn't perfect - but who cares about +/- 2% soc?

**Reply by deadrabbit87** [reply]
I also use evcc with psa car controller.

Why aren't you using the Stellantis API data?

Using the soc as you described wouldn't bring any benefit according to my current implementation.

**Reply by moritzklein** [reply]
I don‘t use the Stellantis API because it’s not reliable. Only working 80% of time. Sometimes I have no connection / car status for 1 week via the App / API. Meatpi wican works 100%.

**Reply by Formstation** [reply]
@moritzklein, I try to do exactly the same thing with my Astra Electric. I have MQTT running, WiCAN Adapter connected with the network, visible in MQTT Explorer, but I don't know how to proceed with the next steps. Could you please publish your WiCAN profile (or screenshots) and evcc.yaml code lines for your vehicle?

**Reply by moritzklein** [reply]
Here is my evcc config for the car:

`vehicles:
  - name: "mokkawican"
    title: "Mokka-WiCAN"
    type: custom

    soc:
      source: mqtt
      topic: "wican/mokka/soc"
      jq: '.soc'
      scale: 1             # No scaling needed if the SoC is already in percentage format
      timeout: 30s # Adjust as needed

    capacity: 46 # Akkukapazität in kWh (optional)
    icon: car # Icon in der Benutzeroberfläche (optional)
    phases: 3 # Die maximale Anzahl der Phasen welche genutzt werden können (optional)`

this, you have to setup mqtt for evcc aswell.... then this will read the SOC from the topic.

WiCAN config:

`Protocol: AutoPID
Automate: -> Custom PIDs: 

Custom Initialisation:
ATZ;ATE0;ATH1;ATSP6;ATS0;ATM0;ATAT1;ATSH6B4;ATFCSH6B4;ATFCSD300000;ATFCSM1;ATCRA694;

Name	Init	PID	      Expression	   Period(ms)	Type	      Send_to
soc             22D4101   (B4*256+B5)/512  1000         MQTT_Topic    wican/mokka/soc`

**Reply by Formstation** [reply]
@moritzklein Thanks a lot! I've made more progress than before. However, things aren't quite working yet. The SoC in MQTT Explorer never changes. In evcc I can't see the SoC at all and I get the notification "Garage: vehicle soc: outdated". Can you see what I did wrong?

<img width="876" height="887" alt="Bildschirmfoto 2025-10-28 um 12 07 20" src="https://github.com/user-attachments/assets/33c413ac-c3e3-4081-a4d6-8d89d8bd8373" />

<img width="401" height="1017" alt="Bildschirmfoto 2025-10-28 um 12 08 53" src="https://github.com/user-attachments/assets/d9540304-58dd-4e29-b76b-a1db783e9963" />

<img width="751" height="646" alt="Bildschirmfoto 2025-10-28 um 12 31 40" src="https://github.com/user-attachments/assets/f861ddbb-3b40-43fc-8cd8-48ae479f48f1" />

<img width="612" height="357" alt="Bildschirmfoto 2025-10-28 um 12 09 27" src="https://github.com/user-attachments/assets/1ba6bde5-01d2-4eb5-b2d7-d93bd8399ca0" />

**Reply by Formstation** [reply]
Here's my yaml code, including mqtt lines:

```
mqtt:
  broker: 192.168.178.69:1883
  user: Astra801
  password: xxxxxxx

vehicles:
- name: opelastra
  type: custom
  title: Opel Astra
  capacity: 51
  soc:
    source: mqtt
    topic: wican/astra/soc
    jq: '.soc'
    timeout: 30s
```

**Reply by Formstation** [reply]
@moritzklein Any idea?

**Reply by Formstation** [reply]
@moritzklein Es war dann doch ganz einfach. Da in den "Vehicle Specific" in der WiCAN Konfiguration unter SOC der "Name: SOC" in Großbuchstaben steht, habe ich in der evcc.yaml lediglich in der Zeile jq: '.soc' auch den SOC groß geschrieben: jq: '.SOC'
Es klappt!
Natürlich mit der bekannten Einschränkung, dass der tatsächliche SoC vom ODB Adapter nur übermittelt wird, wenn die Zündung läuft. Zwar passiert das nur, wenn ich Zuhause ankomme, aber das reicht, denn ab da errechnet sich evcc den Wert selber und zeigt in der evcc-Oberfläche, was er soll. Allerdings bringt er jetzt weiterhin eine Fehlermeldung in orange: "Opel Astra: vehicle soc: outdated (ignored by estimator)" Umgehe diese Meldung, indem ich das timeout des soc in evcc.yaml auf 480m gesetzt habe.

<img width="788" height="655" alt="Bildschirmfoto 2025-11-10 um 14 51 03" src="https://github.com/user-attachments/assets/9999c7be-9320-4f49-b4fa-2611721724dc" />

**Comment by tqma1** [comment]
Would be great if somone can help with the expression for the Battery temperature and kWh available
Manny thanks in advance!

**Reply by meatpiHQ** [reply]
@tqma1 do you these parameters in car scanner app?

**Comment by tqma1** [comment]
Yes, i did.Here the results:

Battery temp
>22D8EF1
6820462D8EF09

now Battery temp is 10
6820462D8EF0B

Ambient temp
>22D4341
6820462D4340A

kWh available 
>22D8651
6940562D865092D

**Reply by tqma1** [reply]
This works now for me:

AT= Ambient temperature
BT= Battery temperature

      {
        "Name": "AT",
        "Init": "ATSH6A2;ATFCSH6A2;ATCRA682;",
        "PID": "22D4341",
        "Expression": "B4",
        "MinValue": "",
        "MaxValue": "",
        "Period": "5000",
        "Type": "Default",
        "Send_to": ""
      },
      {
        "Name": "BT",
        "Init": "ATSH6A2;ATFCSH6A2;ATCRA682;",
        "PID": "22D8EF1",
        "Expression": "B4",
        "MinValue": "",
        "MaxValue": "",
        "Period": "5000",
        "Type": "Default",
        "Send_to": ""
      },
      {
        "Name": "min_mV",
        "Init": "ATSH6B4;ATFCSH6B4;ATCRA694;",
        "PID": "22D86F1",
        "Expression": "[B4:B5]",
        "MinValue": "",
        "MaxValue": "",
        "Period": "5000",
        "Type": "Default",
        "Send_to": ""
      },
      {
        "Name": "max_mV",
        "Init": "ATSH6B4;ATFCSH6B4;ATCRA694;",
        "PID": "22D8701",
        "Expression": "[B4:B5]",
        "MinValue": "",
        "MaxValue": "",
        "Period": "5000",
        "Type": "Default",
        "Send_to": ""
      }

**Reply by tqma1** [reply]
So is this profile correct and should work?

[eUlysse.json](https://github.com/user-attachments/files/20010418/eUlysse.json)

**Reply by meatpiHQ** [reply]
@tqma1 looks good but parameters `"BT" "max_mV" "AT" "BT"` should be added to [params.json ](https://github.com/meatpiHQ/wican-fw/blob/main/.vehicle_profiles/params.json) if they don't already exist

**Reply by deadrabbit87** [reply]
@tqma1 I've created a PR for battery temperature and ambient temperature. `max_mV` and `min_mV ` didn't work on my car. 

https://github.com/meatpiHQ/wican-fw/pull/669

**Reply by evcc-passra** [reply]
at my corsa, it works fine:

<img width="211" height="40" alt="grafik" src="https://github.com/user-attachments/assets/ab129743-1227-4003-975f-eea68e1be793" />

**Comment by zackslash** [comment]
Hey, the e-208 profile is also working great for my Peugeot e-2008.

Does anyone have the PID & expression for the 'Range' value?

**Comment by matthewturner** [comment]
I can confirm the `e-208` profile works on my `Vauxhall Corsa-e 2024` as @slash909uk said above.

However, it's not reporting `SOC`, `SOH` or `HV_V` values **during charging**. I've confirmed the Wican Pro and ECU are both running during this time. Is there anything I can try to enable this during charging? Thanks!

**Reply by evcc-passra** [reply]
Is there already a solution to the problem? I started using WICAN in my Corsa at the weekend and connected it to evcc. I will probably have the same problem.

**Reply by matthewturner** [reply]
I've had to work around it for now in Home Assistant by creating an input_helper and a couple of automations. 

In case this helps, one automation updates the input_helper (current_charge) from Wican when I park up. The other automation watches the charger and when the charging cycle finishes it adds the charge energy to the current_charge input_helper. It's just an estimate, but I've been running it for a week or so and it's been surprisingly accurate and robust.

**Reply by camueller** [reply]
> Is there already a solution to the problem? I started using WICAN in my Corsa at the weekend and connected it to evcc. I will probably have the same problem.

This problem has to be solved by evcc: It gets the inial SOC from WiCAN but has to calculate the SOC during charging - other charging apps do it this way as well. WiCAN has no way of getting the SOC during charging if the CAN bus is not active because the car is switched off.

**Reply by Formstation** [reply]
I solved this for my Opel Astra by creating a custom vehicle in evcc.yaml. The Wican adapter has to write the SoC into an MQTT database that evcc can read. The entry in yaml then looks like this:

```
  mqtt:
  broker: localhost:1883 #if MQTT is running on an other machine, use the correct IP
  user: Astra 
  password: xxxxxxx

vehicles:
- name: opelastra
  type: custom
  title: Opel Astra
  capacity: 50
  soc:
    source: mqtt
    topic: wican/astra/soc
    jq: '.SOC'
    timeout: 720m

```

I set this long timeout to avoid evcc-log-errors like "soc out of date".

When I arrive at home with the Opel, I wait 20 seconds before turning it off to ensure that the Wican adapter transmits the SOC via Wi-Fi to MQTT. From there evcc calculates the SoC during charging.

**Reply by deadrabbit87** [reply]
Can someone confirm this: while charging, i can read only the ODOMETER. 

Therefore i think, the CAN is not sleeping, but maybe there are different PIDs?

**Reply by evcc-passra** [reply]
> I solved this for my Opel Astra by creating a custom vehicle in evcc.yaml. The Wican adapter has to write the SoC into an MQTT database that evcc can read. The entry in yaml then looks like this:
> 
> ```
>   mqtt:
>   broker: localhost:1883 #if MQTT is running on an other machine, use the correct IP
>   user: Astra 
>   password: xxxxxxx
> 
> vehicles:
> - name: opelastra
>   type: custom
>   title: Opel Astra
>   capacity: 50
>   soc:
>     source: mqtt
>     topic: wican/astra/soc
>     jq: '.SOC'
>     timeout: 720m
> ```
> 
> I set this long timeout to avoid evcc-log-errors like "soc out of date".
> 
> When I arrive at home with the Opel, I wait 20 seconds before turning it off to ensure that the Wican adapter transmits the SOC via Wi-Fi to MQTT. From there evcc calculates the SoC during charging.

i did it in the same way, but evcc doesn't calculate a new SOC during charging :-(

**Reply by deadrabbit87** [reply]
@evcc-passra you have to set interpolate soc during charge: 

<img width="774" height="128" alt="grafik" src="https://github.com/user-attachments/assets/1ae06e9d-7416-426c-a190-70a2689da689" />

**Reply by deadrabbit87** [reply]
Here i found the Frame IDs for SOC during charge: 
https://github.com/evcc-io/evcc/discussions/13685#discussioncomment-13914257

All are working on my car. Except for the odometer (mileage). 

With frame ID `0x552` i could not get any realistic mileage.

**Reply by deadrabbit87** [reply]
@meatpiHQ I’m currently testing the IDs mentioned above. My first impression is that they work correctly even when the ignition is off.

However, there is an issue with the mileage data: these messages are only broadcast at the start of a charging session.

Given this, should I proceed with refactoring the current vehicle profile using these IDs and provide a PR?

**Reply by meatpiHQ** [reply]
@deadrabbit87 yes please do create a PR.

**Comment by tqma1** [comment]
As Code for your yaml:
estimate: true # Interpolation aktivieren

**Reply by evcc-passra** [reply]
ok, thank you, i will test it soon :-)

**Reply by evcc-passra** [reply]
Today I tested the calculation, but unfortunately it doesn't work.
EVCC reports "Corsa Elektro: charger soc: outdated", but does not start the calculation. The charge level remains at 76% as at the start, and the charging time is reset every 2 minutes to the originally calculated 34 minutes.
Could it be because the WiCan does not go offline during charging?
<img width="245" height="71" alt="grafik" src="https://github.com/user-attachments/assets/04eae0c0-3a8e-4ed0-ae36-9a90e15acdf4" />

**Comment by hamza-sdr** [comment]
hey, has anyone managed to find the PIDs for cell voltages?

**Comment by petraveller25** [comment]
hello, does anyone able to read the battery current from PID? from car scanner it gives the log below, actual battery current is around 2.69A. What expressions should be put in, in order to read correctly? thanks
[log_battery_current_low.txt](https://github.com/user-attachments/files/25531512/log_battery_current_low.txt)

>22D816
6940762D81600012CAF

**Reply by hamza-sdr** [reply]
hello, I have a Citroen AMI (which is also Stellantis) and I use this expression for the current : ((B6*256+B7)-11264)/64 

<img width="589" height="931" alt="image" src="https://github.com/user-attachments/assets/eacef318-764e-4152-b6cf-0fcd5cd41f2e" />

I notice that there are the cell voltages in the logs, do you know what was its value so I can try to fine their expression ? Also if you have logs for the temperatures with their value that would be very helpful.

**Reply by petraveller25** [reply]
hello 7amza-s, thanks for the quick response, let me try with your expression. My PID 22D8161 data always comes with the 694 header, I assume that the PID init ATSH694; will take care of that?

the cell voltages readings as follows:
PID: 22D8151
PID init: ATSH6B4;ATFCSH6B4;ATCRA694;
Expression: [B4:B5]/16
Unit: V
Class: voltage
Min value: 200
Max value: 480
Period(ms): 5000

the cell temperature readings as follows:
PID: 22D8EF1
PID init: ATSH6A2;ATFCSH6A2;ATCRA682;
Expression: B4
Unit: V
Class: temperature
Min value: (empty)
Max value: (empty)
Period(ms): 5000

I hope it helps.

**Reply by petraveller25** [reply]
by the way, what software are you using to have that car scanner log parser? thanks

**Reply by petraveller25** [reply]
by the way, what software are you using to have that car scanner log parser? thanks

**Reply by hamza-sdr** [reply]
thanks @petraveller25 !! I use this https://meatpihq.github.io/cs_log_parser/

**Reply by petraveller25** [reply]
@7amza-s thank you!

**Comment by AndyMu97** [comment]
Hi all,

I was able to setup WiCAN for charging with Home Assistant and EVCC add-on with my Opel Mokka-e.
The issue with the SOC update during charging in "passive mode" was solved following this comment: https://github.com/meatpiHQ/wican-fw/issues/543#issuecomment-3902397408

The only challenge I have now is: for EVCC I need to get an information from the car if it is connected.
Is there a chance to get this information via a Custom Filter to MQTT? I have no idea which CAN ID / Bit or PID does have this information.

Does anyone solved this "connect status" issue?

**Comment by evcc-passra** [comment]
evcc should get the information "car connected" from the Wallbox, not from the car. My Corsa doesn't send this Information to evcc

**Reply by AndyMu97** [reply]
Hi, the wallbox is only able to see that there is a connection, but not with which car.
Evcc identifies the connected car by verifying all cars and if they are connected, and normally only one car should tell evcc it is connected.
For the moment I‘m fine with a fixed configured car in evcc, but maybe in a few months I will have a second car.

**Comment by evcc-passra** [comment]
I authorise my cars (Corsa, EV4, Mini) at the wallbox using an special RFID chip for every car; this tells the wallbox – and therefore EVCC as well – which car is plugged in and allows evcc to match the SOC.

**Reply by AndyMu97** [reply]
This solution is working well, and I also had it with my Tinkerforge Wallbox. But I wanted to plug in and do nothing else, that’s why I have home automation 😁
I will record the logs of the CAN monitor with MicroSD and ask Gemini to identify the frame/bit when I disconnect and connect.
Will keep you updated.
